Codec/Compression/Zlib/Internal.hs

```diff
@@ -586,13 +587,25 @@ decompressST :: Stream.Format -> DecompressParams -> DecompressStream (ST s)
 --
 decompressIO :: Stream.Format -> DecompressParams -> DecompressStream IO
 
-decompress   format params = foldDecompressStreamWithInput
-                               L.Chunk (const L.Empty) throw
-                               (decompressStreamST format params)
+decompress   format params = either throw id . safeDecompress format params
```

Comment:
I'm against this change.

Just adding `safeDecompress` is OTOH fine and welcome.

Reply:
Can you explain why you're against this change? Based on [this Twitter thread](https://twitter.com/taylorfausak/status/1104416216905584640), I'm assuming it's because this new version of `decompress` is not as lazy as the original one.